src/components/BlogPostPreview.astro

```diff
@@ -1,53 +1,51 @@
 ---
 import { englishMonths, spanishMonths } from '../lib/dates';
-import type { Post } from '../types';
+import type { DualPost } from '../types';
 
 export interface Props {
-  post: Post;
+  post: DualPost;
 }
 
 const { post } = Astro.props as Props;
-
-const createdAt = new Date(post.created_at);
 ---
 
 <div class='shadow-lg md:rounded-2xl bg-white border-t-4 border-t-indigo-500'>
   <div class='flex flex-col md:flex-row md:items-center md:divide-x'>
     <a
       class='p-7 hover:bg-indigo-50 cursor-pointer transition duration-100 rounded-tl-2xl relative flex-grow h-52 overflow-hidden'
-      href={`posts/en/${post.slug}`}
+      href={`posts/en/${post.en.slug}`}
     >
       <div
         class='w-full h-full absolute bg-gradient-to-b from-transparent via-transparent to-gray-100 left-0 top-0 rounded-tl-2xl  transition duration-200'
       >
       </div>
-      <h1 class='text-2xl font-medium text-gray-900'>{post.title}</h1>
-      <div class='text-gray-600 mt-2' set:html={post.content}></div>
+      <h1 class='text-2xl font-medium text-gray-900'>{post.en.title}</h1>
+      <div class='text-gray-600 mt-2' set:html={post.en.content}></div>
     </a>
     <a
       class='p-7 hover:bg-indigo-50 cursor-pointer transition duration-100 rounded-tr-2xl relative flex-grow h-52 overflow-hidden'
-      href={`posts/es/${post.slug}`}
+      href={`posts/es/${post.es.slug}`}
```

Comment:
the two languages have different slugs for each post, so we don't really need the `es/` or `en/` disambiguator, i lean towards removing it, for cleaner urls.